WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewController.swift

```diff
@@ -0,0 +1,180 @@
+import UIKit
+import Yosemite
+import WordPressUI
+import Combine
+
+/// Displays a list of settings for the user to choose to bulk update them for all variations
+///
+final class BulkUpdateViewController: UIViewController {
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    private let viewModel: BulkUpdateViewModel
+
+    private var subscriptions = Set<AnyCancellable>()
+
+    init(viewModel: BulkUpdateViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureTableView()
+        configureViewModel()
+    }
+
+    /// Setup receiving updates for data changes and actives the view model
+    ///
+    private func configureViewModel() {
+        viewModel.$syncState.sink { [weak self] state in
+            guard let self = self else { return }
+
+            switch state {
+            case .notStarted:
+                fallthrough
+            case .syncing:
+                self.displayPlaceholder()
+            case .synced:
+                fallthrough
```

Comment:
I also think it would be more clear if we call `self.displayPlaceholder()` here.

Reply:
I have restructured the switch to be more clear:

            case .synced, .error:
                self.removeGhostContent()